cpp/ql/src/Likely Bugs/Memory Management/AllocaInLoop.ql

```diff
@@ -21,7 +22,15 @@ Loop getAnEnclosingLoopOfStmt(Stmt s) {
 }
 
 from Loop l, FunctionCall fc
-where getAnEnclosingLoopOfExpr(fc) = l
-  and fc.getTarget().getName() = "__builtin_alloca"
-  and not l.(DoStmt).getCondition().getValue() = "0"
-select fc, "Stack allocation is inside a $@ and could lead to overflow.", l, l.toString()
+where
+  getAnEnclosingLoopOfExpr(fc) = l and
+  (
+    fc.getTarget().getName() = "__builtin_alloca"
+    or
+    (
+      (fc.getTarget().getName() = "_alloca" or fc.getTarget().getName() = "_malloca") and
+      fc.getTarget().getADeclarationEntry().getFile().getBaseName() = "malloc.h"
```

Comment:
Can you confirm that this check for `malloc.h` works as intended on a real-world snapshot? I don't know how the Microsoft libraries are structured, but glibc typically doesn't declare any functions in the user-included headers; those headers are just shallow wrappers that include implementation-specific headers, where the functions are actually declared.

Reply:
That's a good point, and I already checked Microsoft's headers.  The _alloca and _malloca functions are indeed declared in malloc.h, so we're good.